ansible_collections/arista/avd/roles/eos_validate_state/python_modules/ip_reachability/avdipreachabilitytest.py

```diff
@@ -0,0 +1,112 @@
+                    continue
+
+                for peer_ethernet_interface in peer_ethernet_interfaces:
+                    if ethernet_interface["peer_interface"] == peer_ethernet_interface["name"]:
+                        src_ip = format(ipaddress.ip_interface(ethernet_interface["ip_address"]).ip)
+                        dst_ip = format(ipaddress.ip_interface(peer_ethernet_interface["ip_address"]).ip)
+                        self.test = (
+                            f"Source: {self.device}_{ethernet_interface['name']} - Destination:"
+                            f" {ethernet_interface['peer']}_{ethernet_interface['peer_interface']}"
+                        )
+                        self._anta_test.append(
+                            {
+                                "VerifyReachability": {
+                                    "template_params": [{"src": src_ip, "dst": dst_ip}],
+                                    "result_overwrite": {"categories": self.categories, "description": self.description, "custom_field": self.test},
+                                }
+                            }
+                        )
+
+        return {self._anta_module: self._anta_test}
+
+
+class AvdLoopback0ReachabilityTests(AvdBaseTest):
+    """
+    AvdLoopback0ReachabilityTests class derived from AvdBaseTest for Loopback0 reachability related tests.
+    """
+
+    def __init__(self, task_vars: dict):
+        super().__init__(task_vars)
+        self._anta_test = []
+        self._anta_module = "anta.tests.connectivity"
+        self.categories = ["Loopback0 Reachability"]
+        self.description = "Loopback0 Reachability"
+
+    def generate_test_definition(self) -> dict:
+        """
+        Generates the ANTA test definition for the Loopback0 reachability tests.
+
+        Returns:
+            test_definition (dict): ANTA test definition.
+
+        """
+        loopback_mappings = []
+
+        fabric_name = get(self.task_vars, "fabric_name")
+        groups = get(self.task_vars, "groups")
+        node_type = get(self.task_vars, "type")
+
+        if any(v is None for v in [fabric_name, groups, node_type]):
+            self.display.error(f"{self.device}: 'fabric_name', 'groups' or 'type' is missing from structured_config. {self.__class__.__name__} is skipped.")
+            return {}
+
+        for node in groups[fabric_name]:
+            if (loopback_interfaces := get(self.task_vars, f"hostvars.{node}.loopback_interfaces")) is None:
+                self.display.error(f"{node}: 'loopback_interfaces' is missing from structured_config.")
+                continue
+
+            for loopback_interface in loopback_interfaces:
+                if loopback_interface["name"] == "Loopback0":
+                    loopback_mappings.append((node, format(ipaddress.ip_interface(loopback_interface["ip_address"]).ip)))
+
+        if node_type == "l3leaf":
```

Comment:
This should be refactored at some point. We have an open issue on that. We should not read any vars from eos_designs like `type`, `fabric_name` etc. and we should also not use Ansible's `groups`.